docs/gettinginvolved/plugins.rst

```diff
@@ -0,0 +1,93 @@
+
+To test that your implementation is correct you can instantiate a generic test suite that makes sure your class correctly implements the interface:
+
+.. code-block:: python
+
+    from flocker.node.agents.test.test_blockdevice import make_istatechange_tests
+
+    def api_factory(test):
+        # Return an instance of your IBlockDeviceAPI implementation class, given
+        # a twisted.trial.unittest.TestCase instance.
+
+    # Smallest volume to create in tests, e.g. 1GiB:
+    MIN_ALLOCATION_SIZE = 1024 * 1024 * 1024
+
+    # Minimal unit of volume allocation, e.g. 1MiB:
+    MIN_ALLOCATION_UNIT = 1024 * 1024
+
+    class YourStorageTests(make_istatechange_tests(
+        api_factory, MIN_ALLOCATION_SIZE, MIN_ALLOCATION_UNIT)):
+        """
+        Tests for your storage.
+        """
+
+You can run these tests with ``trial`` test runner provided by `Twisted <https://twistedmatrix.com>`_, one of Flocker's dependencies:
+
+.. code-block:: sh
+
+    $ trial yourstorage.test_yourstorage
+
+
+Implementing storage plugins
+============================
+
+Once you've implemented your storage backend you'll want to allow Flocker users to utilize your package.
+The basic implementation strategy is that your users install a Python package with your backend implementation on all Flocker nodes:
+
+.. code-block:: sh
+
+    $ /opt/flocker/bin/pip install https://example.com/your/yourstorageplugin-1.0.tar.gz
+
+You can also provide RPMs or DEBs that have same effect of installing a new Python package.
+
+Once your users have installed the package, they will write a ``/etc/flocker/agent.yml`` whose ``backend`` key in the ``dataset`` section is the importable name of the Python package you've installed.
+All other sub-keys of the ``dataset`` section will be passed to a function you must implement (see below) and can be used to configure the resulting ``IBlockDeviceAPI`` instance.
+Typical parameters are authentication information or server addresses; whatever is necessary to configure your class.
+
+For example, if you installed a Python package importable ``mystorage_flocker_plugin``, and you require a username and password in order to login to your storage system, you might tell your users to write a ``agent.yml`` that looks like this:
+
+.. code-block:: yaml
+
+   version: 1
+     control-service:
+       hostname: "user.controlserver.example.com"
+     dataset:
+       backend: "mystorage_flocker_plugin"
+       username: "username_for_mystorage"
+       password: "abc123"
+
+Your ``mystorage_flocker_plugin/__init__.py`` module needs to have a ``FLOCKER_BACKEND`` attribute with a ``flocker.node.BackendDescription`` instance, which will include a reference to factory function that constructs a ``IBlockDeviceAPI`` instance.
+The factory function will be called with whatever parameters the ``dataset`` section in ``agent.yml`` is configured with; in the above example that would be ``username`` and ``password``.
```

Comment:
At some point it was discussed that plugins would be able to provide their own `jsonschema`. Make a follow-up issue if you think that there should be one for that.